### PR TITLE
add feature support matrix for PyTorch backend

### PR DESCRIPTION
**Pull request by QiJune**

### CODE REVIEW

File: docs/source/torch/features/feature_combination_matrix.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+| Torch Sampler              | Yes               | Yes        | Yes                        | Yes                   | Yes             | Yes | Yes                       | Yes                       | ---           |                  |                |                        |                       |           |
+| TLLM C++ Sampler           | Yes               | Yes        | Yes                        | No                    | Yes             | No  | No                        | No                        | No            | ---              |                |                        |                       |           |
+| KV Cache Reuse             |                   |            |                            |                       |                 |     |                           |                           |               |                  | ---            |                        |                       |           |
```
Comment (QiJune) [inline docs/source/torch/features/feature_combination_matrix.md:15]: Hi @byshiue , based on you testing, do you know if KV cache reuse can work with other features like overlap scheduler/CUDA graph/attention DP/disaggregated serving/etc? Thanks

File: docs/source/torch/features/feature_combination_matrix.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+| EAGLE-3(Two Model Engine)  | NO                | Yes        | Yes                        | No                    | Untested        | No  | No                        | ---                       |               |                  |                |                        |                       |           |
+| Torch Sampler              | Yes               | Yes        | Yes                        | Yes                   | Yes             | Yes | Yes                       | Yes                       | ---           |                  |                |                        |                       |           |
+| TLLM C++ Sampler           | Yes               | Yes        | Yes                        | No                    | Yes             | No  | No                        | No                        | No            | ---              |                |                        |                       |           |
```
Comment (QiJune) [inline docs/source/torch/features/feature_combination_matrix.md:14]: @dcampora  Could you please help review the TLLM C++ Sampler part? To make sure the latest status is catched up.

File: docs/source/torch/features/feature_combination_matrix.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+| Slide Window Attention     |                   |            |                            |                       |                 |     |                           |                           |               |                  |                | ---                    |                       |           |
+| Logits Post Processor      |                   | Yes        | Yes                        | No                    | Untested        | No  | No                        | No                        | Yes           | Yes              |                |                        | ---                   |           |
+| X-Grammar                  | No                | Yes        | Yes                        | No                    | Untested        | No  | No                        | No                        | Yes           | Yes              |                |                        |                       | ---       |
```
Comment (syuoni) [inline docs/source/torch/features/feature_combination_matrix.md:18]: It would be better to use "Guided Decoding" as the feature name. We could have guided decoding backend other than XGrammar like LLGuidance (https://github.com/NVIDIA/TensorRT-LLM/pull/5214)

Updated feature combination for Guided Decoding:
* Overlap Scheduler: No
* CUDA Graph: Yes
* Attention Data Parallelism: Yes
* Disaggregated Serving: Untested
* Chunked Prefill: Yes
* MTP: No
* EAGLE-3(One Model Engine): No
* EAGLE-3(Two Model Engine): No
* Torch Sampler: Yes
* TLLM C++ Sampler: Yes
* KV Cache Reuse: Yes
* Slide Window Attention: Yes
* Logits Post Processor: Yes

File: docs/source/torch/features/feature_combination_matrix.md (hunk 1 above)
Comment (byshiue) [reply docs/source/torch/features/feature_combination_matrix.md:15]: I have verified `kv cache reuse` with `overlap_scheduler` and `attention DP`. 

For `kv cache reuse` + `CUDA graph`, it could work, but the there are few performance issues. In fact, when `kv cache reuse` is enabled and reuse some caches, this means that there are few `context request` in the batch and hence we never use `CUDA graph` in such case.

For `disaggregated serving`, I don't verify.

For `Eagle3`, we are asked to disable the `kv cache reuse` now and I haven't tried to enable the `kv cache reuse` when we use `Eagle3`.

Comment (QiJune) [reply docs/source/torch/features/feature_combination_matrix.md:15]: + @lfr-0531  and @mikeiovine  to comment, do you know if kv cache reuse can work with Eagle-3 or not?

File: docs/source/torch/features/feature_combination_matrix.md (hunk 3 above)
Comment (QiJune) [reply docs/source/torch/features/feature_combination_matrix.md]: Thanks for your inputs, I updated the table.

File: docs/source/torch/features/feature_combination_matrix.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+| TLLM C++ Sampler           | Yes               | Yes        | Yes                        | Yes                   | Yes             | No  | No                        | No                        | No            | ---              |                |                        |                       |                 |
+| KV Cache Reuse             | Yes               | Yes        | Yes                        | Untested              |                 | Yes | Yes                       | No                        | Yes           | Untested         | ---            |                        |                       |                 |
+| Slide Window Attention     | Untested          |            |                            |                       |                 |     |                           |                           |               |                  | WIP            | ---                    |                       |                 |
```
Comment (QiJune) [inline docs/source/torch/features/feature_combination_matrix.md:16]: @ming-wei  @PerkzZheng  Could you please review the Slide Window Attention part? Thanks

Comment (PerkzZheng) [reply docs/source/torch/features/feature_combination_matrix.md:16]: that looks good to me. it is not ready until Ming' PR is merged.

File: docs/source/torch/features/feature_combination_matrix.md (hunk 1 above)
Comment (dcampora) [reply docs/source/torch/features/feature_combination_matrix.md]: I would imagine that `kv cache reuse` is orthogonal to the choice of sampler, I am not aware of any code in the sampler where kv cache reuse is affected.

File: docs/source/torch/features/feature_combination_matrix.md (hunk 2 above)
Comment (dcampora) [reply docs/source/torch/features/feature_combination_matrix.md]: I would imagine that kv cache reuse is orthogonal to the choice of sampler, I am not aware of any code in the sampler where kv cache reuse is affected, so both Torch Sampler and TRTLLM Sampler should be supported. It would be better to have a test case of course.

File: docs/source/torch/features/feature_combination_matrix.md (hunk 4 above)
Comment (ming-wei) [reply docs/source/torch/features/feature_combination_matrix.md:16]: We haven't tested KV cache reuse + MTP/EAGLE-3. Let's put "Untested" there?

Comment (QiJune) [reply docs/source/torch/features/feature_combination_matrix.md]: Hi @ming-wei , I make KV cache reuse + MTP as untested. @lfr-0531  told me kv cache reuse + EAGLE-3 is supported.

File: docs/source/torch/features/feature_combination_matrix.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+| TLLM C++ Sampler           | Yes               | Yes        | Yes                        | Yes                   | Yes             | No       | No                        | No                        | No            | ---              |                |                        |                       |                 |
+| KV Cache Reuse             | Yes               | Yes        | Yes                        | Untested              | Untested        | Untested | Yes                       | No                        | Yes           | Yes              | ---            |                        |                       |                 |
+| Slide Window Attention     | Yes               | Yes        | Yes                        | Untested              | Untested        | Untested | Untested                  | Untested                  | Yes           | Yes              | WIP            | ---                    |                       |                 |
```
Comment (laikhtewari) [inline docs/source/torch/features/feature_combination_matrix.md:16]: For WIP items, can we link to a draft PR or branch? 

Does not need to block merging this because this looks great

**Review by laikhtewari — APPROVED** [review]

File: docs/source/torch/features/feature_combination_matrix.md (hunk 5 above)
Comment (QiJune) [reply docs/source/torch/features/feature_combination_matrix.md:16]: After discussing with @ming-wei , we will update the PR link later.
